framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md

```diff
@@ -0,0 +1,30 @@
+# FunctorElementalGradientAux
+
+!syntax description /AuxKernels/FunctorElementalGradientAux
+
+## Overview
+
+This `AuxKernel` outputs a vector `AuxVariable` given by scaling the gradient of a functor scaled by a material property and/or another functor.
+
+!alert note title=Use in Optimization module
+This `AuxVariable` is used to linearize a nonlinear material property in the adjoint equation.  
+
+## Example Input File Syntax
+
+The use of this `AuxKernel` to compute the adjoint for a temperature dependent material is shown in this example:
+
+!listing modules/combined/test/tests/invOpt_nonlinear/adjoint.i block=AuxKernels
+
+where the `AuxVariable` dDdTgradT is defined as a constant monomial vector:
+
+!listing modules/combined/test/tests/invOpt_nonlinear/adjoint.i block=AuxVariables/dDdTgradT  
```

Comment:
I didn't know you could do listings from modules in framework doc

Reply:
This is a little dangerous since trying to build from `framework/doc/` will cause a failure. I think you can use the [IfElse extension](https://mooseframework.inl.gov/python/MooseDocs/extensions/ifelse.html) to avoid this.

Reply:
there's over 40 instances of that in framework/doc/content

Reply:
Interesting, I guess the the `git_ls_files` works all over the repo.